src/PrMerger.js

```diff
@@ -35,17 +57,37 @@ class PrMerger {
             try {
                 const rawPr = this._todo.shift();
 
+                scanResult.awakePrs.push(rawPr);
+
                 if (rawPr.labels.some(el => el.name === Config.ignoredByMergeBotsLabel())) {
                     this._ignored++;
                     Logger.info(`Ignoring PR${rawPr.number} due to ${Config.ignoredByMergeBotsLabel()} label`);
                     continue;
                 }
 
+                const clearedForMerge = rawPr.labels.some(el => el.name === Config.clearedForMergeLabel());
+
+                // There are usually few 'cleared' PRs (i.e., ready for merge), so do not ignore them.
+                // This should help handle sutiations where some of such 'cleared' PRs were updated just after
+                // this scan has been started. Other (non-cleared) PRs (that are not going to be merged now anyway)
+                // can be ignored until the next scan.
+                // TODO: also handle a situation when the PR becomes 'cleared' just after this scan began.
+                if (!clearedForMerge && _LastScan && _LastScan.isStillUnchanged(rawPr)) {
+                    Logger.info(`Ignoring PR${rawPr.number} because it has not changed since the previous scan of ${_LastScan.scanDate.toISOString()}`);
+                    this._ignoredAsUnchanged++;
+                    continue;
+                }
+
                 const result = await MergeContext.Process(rawPr, somePrWasStaged);
                 assert(!somePrWasStaged || !result.prStaged());
                 somePrWasStaged = somePrWasStaged || result.prStaged();
                 if (result.delayed() && (minDelay === null || minDelay > result.delayMs()))
                     minDelay = result.delayMs();
+
+                // delayed PRs will be processed (when the delay expires) even if not updated
+                if (result.delayed())
+                    scanResult.awakePrs = scanResult.awakePrs.filter(el => el.number !== rawPr.number);
+
```

Comment:
```suggestion
                    currentScan.forgetPr(rawPr);
```

And please assert, inside the new method, that the given PR _is_ found.